backend/sportsmatch/src/main/java/com/sportsmatch/auth/AuthService.java

```diff
@@ -0,0 +1,45 @@
+package com.sportsmatch.auth;
+
+import com.sportsmatch.dtos.RequestDTO;
+import com.sportsmatch.mappers.UserMapper;
+import com.sportsmatch.models.User;
+import com.sportsmatch.repos.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthService {
+
+  private final UserMapper userMapper;
+  private final JwtService jwtService;
+  private final UserRepository userRepository;
+  private final AuthenticationManager authenticationManager;
+
+  public RequestDTO register(RequestDTO requestDTO) {
+    User user = userMapper.registerToUser(requestDTO);
+    if (user == null) {
+      return RequestDTO.builder().status("Invalid User Data").build();
```

Comment:
It would be better to throw a HttpStatusException that would return status code 400 (bad request)

In this case this line can be deleted because of the automatic validation (see other comments)